src/assets/TypeScriptAsset.js

```diff
@@ -28,9 +31,39 @@ class TypeScriptAsset extends Asset {
 
     // Overwrite default if config is found
     if (tsconfig) {
+      // Files read by TypeScript parser while resolving "extends"
+      let readFiles = [];
+      // https://github.com/Microsoft/TypeScript/blob/62306bc3f9e119dc17b400ff263dad532da6cdb1/scripts/build/upToDate.js#L370
+      const parseConfigHost = {
+        useCaseSensitiveFileNames: true,
+        getCurrentDirectory: () => process.cwd(),
+        readDirectory: file => fs.readdirSync(file),
+        fileExists: file => fs.existsSync(file) && fs.statSync(file).isFile(),
+        readFile: file => {
+          readFiles.push(file);
+          return fs.readFileSync(file, 'utf8');
+        },
+        onUnRecoverableConfigFileDiagnostic: () => undefined
+      };
+      // Get directory of tsconfig.json
+      const tsconfigDir = path.dirname(
+        await config.resolve(this.name, ['tsconfig.json'])
+      );
+      // Parse contents of tsconfig.json with TypeScript API to resolve "extends"
+      const parsedTsconfig = typescript.parseJsonConfigFileContent(
+        tsconfig,
+        parseConfigHost,
+        tsconfigDir
+      );
+      // Add resolved files to watch list
+      for (const file of readFiles) {
+        if (file.endsWith('/tsconfig.json')) {
```

Comment:
This won't work with windows afaik.

Windows would be `\tsconfig.json`

Reply:
The typescript API outputs `/` separated path, so it passes the test and is added as dependency. I guess I have to convert the path from typescript with `path.resolve` (if it actually converts, though).

For reference, `b.entryAsset.dependencies`:
```
2018-10-04T21:29:53.7376659Z Map {
2018-10-04T21:29:53.7377181Z   'D:\\a\\1\\s\\test\\integration\\typescript-extends\\subdir\\tsconfig.json' => { name: 'D:\\a\\1\\s\\test\\integration\\typescript-extends\\subdir\\tsconfig.json',
2018-10-04T21:29:53.7377379Z   includedInParent: true },
2018-10-04T21:29:53.7377582Z   'D:/a/1/s/test/integration/typescript-extends/tsconfig.json' => { name: 'D:/a/1/s/test/integration/typescript-extends/tsconfig.json',
2018-10-04T21:29:53.7377734Z   includedInParent: true },
2018-10-04T21:29:53.7377858Z   'D:\\a\\1\\s\\test\\.babelrc' => { name: 'D:\\a\\1\\s\\test\\.babelrc', includedInParent: true },
2018-10-04T21:29:53.7378014Z   'D:\\a\\1\\s\\package.json' => { name: 'D:\\a\\1\\s\\package.json', includedInParent: true } }
```